tests/unit/api/test_targets.py

```diff
@@ -62,28 +121,27 @@ def test_post(self, monkeypatch, test_client, token_headers):
                 **target["info"]["custom"],
             }
             expected_payload["targets"].append(target)
-
         assert mocked_repository_metadata.apply_async.calls == [
             pretend.call(
                 kwargs={
                     "action": "add_targets",
-                    "payload": expected_payload,
+                    "payload": payload,
                 },
                 task_id=fake_task_id,
                 queue="metadata_repository",
                 acks_late=True,
             )
         ]
 
-    def test_post_without_add_task_id_to_custom(
+    def test_post_with_add_task_id_to_custom_false(
```

Comment:
This test is repeating what you do in `test_post_with_add_task_id_to_custom`.
Either you can remove the false case from `test_post_with_add_task_id_to_custom` or remove this test.
Another possible solution will be to keep this test with its original title but remove `add_task_id_to_custom` from the payload.
I am not sure if we have such a test without any value for `add_task_id_to_custom`.